### PR TITLE
First pass

### PR DESCRIPTION
**Pull request by cramertj**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -151,51 +153,48 @@
...
                 <section>
-                    However, a Future on it's own cannot execute. We need something to
-                    get it to do some actual work.<br>
-                    <img src="assets/why.gif"></img>
+                    In Rust:<br><br>
+                    <ul>
+                        <li>Future is a trait</li>
+                        <li>Does nothing unless passed to an executor and run</li>
+                        <li>Methods create wrapper types which run their children</li>
+                        <li>One allocation per top-level task</li>
```
Comment (mgattozzi) [inline index.html]: How does Rust do this? I want to be able to explain this properly

File: index.html [hunk 2]
```diff
@@ -151,51 +153,48 @@
...
                 </section>
-                
                 <section>
-                    However, a Future on it's own cannot execute. We need something to
-                    get it to do some actual work.<br>
-                    <img src="assets/why.gif"></img>
+                    In Rust:<br><br>
+                    <ul>
+                        <li>Future is a trait</li>
+                        <li>Does nothing unless passed to an executor and run</li>
```
Comment (mgattozzi) [inline index.html]: So lazy evaluation then? Sort of like Haskell?

File: index.html [hunk 3]
```diff
@@ -151,51 +153,48 @@
...
+                        <li>Methods create wrapper types which run their children</li>
```
Comment (mgattozzi) [inline index.html]: Meaning
I have a future with a future in it and I run it via await first.

right?

(I now understand how the use of yield in await works now and wow that's a smart use of it)

File: index.html [hunk 4]
```diff
@@ -151,51 +153,48 @@
...
                 </section>
-
                 <section>
-                    How do we do that? Polling it with an Event Loop!
+                    How are Rust futures run?<br><br>
+                    <ul>
+                        <li>Future spawned onto executor</li>
+                        <li>Executor polls a Future</li>
+                        <li>Future says "I'm not ready yet!"</li>
+                        <li>Executor goes to sleep until Future sends a wakeup</li>
```
Comment (mgattozzi) [inline index.html]: tempted to put a back to sleep when being woken up reaction image I know I have buried somewhere here which perfectly encapsulates this feeling

File: index.html [hunk 5]
```diff
@@ -215,16 +214,18 @@
...
-}</code></pre>
-                </section>
-                <section>
-                    When we poll a Future we're asking it whether it has computed the value yet!
-                </section>
-                <section>
-                    If it is ready we get the value we wanted otherwise it is scheduled to continue to run. We can ask again later if it's ready!
+}
+
+trait Future {
```
Comment (mgattozzi) [inline index.html]: Ah yeah this is a good addition here.

File: index.html [hunk 6]
```diff
@@ -234,49 +235,33 @@
...
                     Let's do a concrete example together!<br>
                     <img src="assets/together.jpg"></img>
                 </section>
                 <section>
-                    Open up that Cargo.toml and change it to look something like this:
-                    <pre><code>cargo-features=["edition"]
+                    <pre><code>
+ $ cargo new my_executor --bin
+ $ cd my_executor
+ $ vim Cargo.toml # emacs is fine, too, I guess
```
Comment (mgattozzi) [inline index.html]: lol I agree, but we should put `# or the editor you're comfortable with`
Starting the the 5th editor holy war is not how I want my day to go

File: index.html [hunk 7]
```diff
@@ -315,7 +300,7 @@
                 <section>
                     <pre><code>struct Exec;
 impl Executor for Exec {
-    fn spawn_obj(&amp;mut self, _obj: FutureObj&lt;'static, ()&gt;) 
+    fn spawn_obj(&amp;mut self, _obj: FutureObj&lt;'static, ()&gt;)
```
Comment (mgattozzi) [inline index.html]: This is what I get when I don't use my linux setup to catch trailing whitespace.
Thanks for getting that

File: index.html [hunk 8]
```diff
@@ -103,45 +111,39 @@
             <!-- History tiiiiiiime -->
             <section>
                 <section>
-                    We can't understand the Future if we don't understand the past<br>
+                    The past and present of Future<br>
```
[the remark this answers is not in the document]
Comment (cramertj) [reply index.html]: Done.

File: index.html [hunk 9]
```diff
@@ -103,45 +111,39 @@
...
-                    <ul>
-                        <li>This PR is older than most people have been Rust Programmers</li>
-                        <li>https://github.com/rust-lang/rust/pull/10965</li>
-                    </ul>
                 </section>
                 <section>
-                    However, all was not well before Rust 1.0. libgreen was fundamentally at
-                    odds with Rust's ownership model 
-                    <img src="assets/atodds.jpg"></img>
+                    Stack composed of runtime linked list of stacks:
```
[the remark this answers is not in the document]
Comment (cramertj) [reply index.html]: lol

File: index.html (hunk 1 above)
Comment (cramertj) [reply index.html]: There's a single future contained inside of a task, and that future is a composition of all of its child futures, so there's only one allocation needed.

File: index.html (hunk 2 above)
Comment (cramertj) [reply index.html]: yup

File: index.html (hunk 3 above)
Comment (cramertj) [reply index.html]: yeah, that's the idea. it's more explicit in the combinator methods in futures-rs which literally take `self` by value and wrap it in a new future type, similar to how iterator combinators work.

File: index.html (hunk 4 above)
Comment (cramertj) [reply index.html]: resist the temptation, michael. be strong for all of us.

File: index.html (hunk 5 above)
Comment (cramertj) [reply index.html]: ty!

File: index.html (hunk 6 above)
Comment (cramertj) [reply index.html]: heh, I've deleted this all and replaced it w/ the appropriate GH template link

File: index.html (hunk 7 above)
Comment (cramertj) [reply index.html]: np!

**Comment by mgattozzi** [comment]
Hey @cramertj I'm gonna merge this then make some stylistic changes before today's training at Mozilla :D

**Pull request merged**